src/setup-dev-server.js

```diff
@@ -52,43 +52,46 @@ module.exports = function setupDevServer(app, config, cb) {
         new webpack.NoEmitOnErrorsPlugin(),
     );
 
-    // dev middleware
+    // Launch the client webpack build, wait for it to finish, and then kick off
+    // the server build.  Grab the updated clientManifest and serverBundle when
+    // they finish and call update() to update the app
     console.error('Launching client webpack build');
     const clientCompiler = webpack(clientConfig);
-    const devMiddleware = webpackDevMiddleware(clientCompiler, { outputFileSystem: mfs });
-    app.use(devMiddleware);
-    clientCompiler.plugin('done', (stats) => {
+    let serverCompiler;
+    app.use(webpackDevMiddleware(clientCompiler, { outputFileSystem: mfs }));
+    app.use(webpackHotMiddleware(clientCompiler, { heartbeat: 5000 }));
+
+    clientCompiler.hooks.done.tap('setup-dev-server', (clientStats) => {
         console.error('Completed client webpack build');
-        const json = stats.toJson();
-        json.errors.forEach(err => console.error(err));
-        json.warnings.forEach(err => console.warn(err));
-        if (json.errors.length) {
+        if (clientStats.toJson().errors.length) {
             return;
         }
+
+        // Update the manifest for the app
         clientManifest = JSON.parse(mfs.readFileSync(config.clientManifest, 'utf-8'));
-        update();
-    });
 
-    // hot middleware
-    app.use(webpackHotMiddleware(clientCompiler, { heartbeat: 5000 }));
+        // Launch the sever webpack build
+        if (!serverCompiler) {
```

Comment:
I need to try to refresh my memory on the need for this, pulled it from a stash from months ago when I first took a stab at this.  I lifted this back to the root again quickly and seems there's some sort of deadlock once the builds finish when trying to render.  Will step through a debugger to confirm the underlying issue.